client/my-sites/stats/hooks/use-notice-visibility-query.ts

```diff
@@ -40,16 +59,22 @@ export async function queryNotices( siteId: number | null ): Promise< Notices >
 		path: `/sites/${ siteId }/jetpack-stats-dashboard/notices`,
 	} );
 
-	return processConflictNotices( payload );
+	return { ...DEFAULT_NOTICES_VISIBILITY, ...payload };
 }
 
 export default function useNoticeVisibilityQuery( siteId: number | null, noticeId: string ) {
 	return useQuery( {
 		queryKey: [ 'stats', 'notices-visibility', siteId ],
-		queryFn: () => queryNotices( siteId ),
+		queryFn: () => queryNotices( siteId ).then( processConflictNotices ),
 		select: ( payload: Record< string, boolean > ): boolean => !! payload?.[ noticeId ],
-		staleTime: 1000 * 30, // 30 seconds
-		retry: 1,
-		retryDelay: 3 * 1000, // 3 seconds
+		...QUERY_OPTIONS,
+	} );
+}
+
+export function useNoticesVisibilityQueryRaw( siteId: number | null ) {
```

Comment:
👋🏻 I want to make a quick suggestion here. RQ offers an API to format data exactly how you want it via the `select` option ([docs](https://tanstack.com/query/latest/docs/react/reference/useQuery)). Instead of introducing a new custom query hook (which operates the same data but uses a different query key hence does separate requests) you could remove the `processConflictNotices` call from `useNoticeVisibilityQuery` (so it operates on "raw" data) and shift the processing to `select` (either in consumer land directly or in a separate custom hook if you need processed data in several different places). This way round you don't have to introduce a new custom query hook because the base query already processes the data it fetches. What do you think?

Reply:
Hi @flootr Thanks for the suggestion! It makes total sense to me. I'll make the change. Thanks again ❤️

Reply:
Refactored in 4aa3789cc7